### DISCUSSION TITLE
Automatic registration of windows Wazuh clients with SO (REDUX)

### DISCUSSION THREAD

**Discussion by isaacgolding** [Security-Onion-Solutions/securityonion#8047] [Unsupported Versions] [answered]
So one of the features that I've been working on enabling is the enabling of automatic registration for windows wazuh clients.   There doesn't seem to be any current 2.3 version directions that I can find on the topic.

I can manually setup each wazuh agent by manually importing the key every time but it tends to be time consuming on a large scale.  

I found some stuff from the prior builds...but it seems to be missing something.   Any suggestions or updated setup directions to follow would be greatly appreciated.

REF:  https://groups.google.com/g/security-onion/c/y0liAI3um0Y

**Comment by dougburks** [comment]
Have you looked at this section of our documentation?
https://docs.securityonion.net/en/2.3/wazuh.html#automated-deployment

**Comment by isaacgolding** [comment, marked as answer]
Actually I did spend quite a bit of time in the docs and was just going around in circles.    What I figured out later yesterday was a much simpler process that worked for me.
The basic workflow was to open the ports via so-allow for wazuh
then running the following commands (windows install from admin powershell) 

D:\wazuh-agent-3.13.1-1.msi /q ADDRESS="10.23.83.240" AUTHD_SERVER="10.23.83.240"
cd  'C:\Program Files (x86)\ossec-agent'
./agent-auth.exe -m 10.23.83.240

This is working for my particular needs.

**Reply by morohyahaya** [reply]
Hello, is it valid if I have several servers and also how can I place these orders that you quoted automatically on all machines.
